src/esp/scene/SceneNode.cpp

```diff
@@ -3,35 +3,197 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "SceneNode.h"
+#include "SceneGraph.h"
+#include "esp/core/Check.h"
 #include "esp/geo/geo.h"
+#include "esp/sensor/Sensor.h"
 
 namespace Mn = Magnum;
 
 namespace esp {
 namespace scene {
 
-SceneNode::SceneNode(SceneNode& parent)
-    : Mn::SceneGraph::AbstractFeature3D{*this} {
-  setParent(&parent);
-  setId(parent.getId());
+SceneNode::SceneNode() : Mn::SceneGraph::AbstractFeature3D{*this} {
   setCachedTransformations(Mn::SceneGraph::CachedTransformation::Absolute);
   absoluteTransformation_ = absoluteTransformation();
+  // nodeSensorSuite_ and subtreeSensorSuite_ will be released upon SceneNode's
```

Comment:
`// Once created, nodeSensorSuite_ and subtreeSensorSuite_ are features owned by the scene graph node.
No need to release them in the destructor since magnum scene graph will handle it. `

Reply:
Thank you! Let me know if there are any other changes you think I should make.